compass/landice/tests/mismipplus/run_model.py

```diff
@@ -56,17 +72,20 @@ def __init__(self, test_case, name='run_model', subdir=None,
         if suffixes is None:
             suffixes = ['landice']
         self.suffixes = suffixes
+
+        if resolution is None:
+            resolution = 0.0
```

Comment:
How does this work?

Reply:
Yeah this is definitely strange but has to do with the very different way `resolution` is set between `SmokeTest` and `SpinUp` test cases. 

## `SmokeTest` 
So, this last round of commits removes the option to specify the `SmokeTest` resolution in a config file. The problem with the former approach is a user could request a resolution not supported by test case (currently, any value other than `2000m`). With the current code, the supported resolutions are added as individual test cases at the test group level:
https://github.com/andrewdnolan/compass/blob/67b26525f73724c1d6ba0b2f5d7c724427a2c9e0/compass/landice/tests/mismipplus/__init__.py#L19-L21
(This also has the added benefit of the supported resolutions being shown in `compass list`). __So__, for the `SmokeTest` a resolution is always passed to the initialization and in the code above `resolution` will never be `None`. 

## `SpinUp` 
Okay, so now we get to why a "dummy" value of `0.0` is specified. For the `SpinUp` we want to support arbitrary resolution, which should be set in the users config file. The big hiccup here is that we want to use the requested resolution in the directory structure, but the config file which isn't fully parsed until after the step is initialized. Ideally we'd be able to pass `resolution` into the initialization of the `SpinUp` testcase, but that doesn't work. (We can parse the resolution value in the compass repo's config before initialization, but can not parse a user specified config file (e.g. `compass setup -f ....`) prior to the testcases initialization). So, instead we have this dummy value of `0.0` and instead set the value of the `resolution` attribute in the `SpinUp.configure` method so that we have access to all the config options: 
https://github.com/andrewdnolan/compass/blob/67b26525f73724c1d6ba0b2f5d7c724427a2c9e0/compass/landice/tests/mismipplus/spin_up/__init__.py#L49-L83

So, in short the `SmokeTest` knows the value of the `resolution` attribute at the time of initialization, whereas the `SpinUp` needs to parse `resolution` from the config options, so a temporary dummy value is used until the `resolution` attribute is set by the `SpinUp.configure` method. 

Now, you might ask: Why set the dummy value of `0.0` at all? Why not just leave the `resolution` attribute unset for the `SpinUp` testcase until the configure method is called? That's definitely an option, but in talking to Xylar that seems like poor python practice. I guess it's best to initialize all attributes within the `__init__` method. 

Happy to ditch this if need be in order to have a bit clearer but less "proper" python code.

Reply:
Okay, this is fine. I just wanted to make sure it was actually doing something intentional.

Reply:
I added a comment in [f916f60](https://github.com/MPAS-Dev/compass/pull/737/commits/f916f60ccc9d7b8e515ef6a8a775fd71979c5b40) which summarizes the message above to provide a brief explanation to new/future users.